### ISSUE TITLE
Faceted Search shows generic cover image for the product when a combination filter is active

### ISSUE THREAD

**Issue by Maboz** [PrestaShop/PrestaShop#12501]
**Describe the bug**
I have products with combinations, each combination has an attribute (let's name it "pattern"), and specific images for that combination. I created a faceted filter for the "pattern" attribute, so that users can filter on it to see products with that "pattern".
The problem is that when the user filters on a specific "pattern", the products with that attribute are properly shown, but they have their generic cover image instead of the combination image for that pattern.
Imagine flitering on red bags, and you get a list of blue products.

**To Reproduce**
1. Create a pattern attribute, assign 2 values, red and blue
2. Create a product  with 2 combinations, one with pattern red, one with pattern blue
3. assign images properly per combinations, assign the blue image as cover image
3. filter products on red pattern
4. you see a blu product

**Additionnal information**
PrestaShop version: 1.7.5.0
PHP version: 7.2.14

## SPECS

If one filter is selected in FO (let's say Red), we should display the first image of this combination.
If many filters are selected in FO (let's say Blue and black) and a product has these 2 colors, we should display the first image of the default combination in product list.
If no filter are selected in FO, we should display the cover image of the product (issue #12241)

**Comment by khouloudbelguith** [comment]
Hi @Maboz,

In fact, when you filter with the faceted search, the image of the combination by default will be displayed ( if the blue combination is by default when you filter by red/blue, the blue image will be always displayed, if the red combination is by default when you filter by blue, the red image will be displayed.)
I attached a video record.
https://drive.google.com/file/d/1E0suiiQ4asOSTuEUhzBHNRlyO3WGLWhv/view
Thanks to check & feedback.

**Comment by Maboz** [comment]
Yes the video exactly represents the problem, it's quite confusing for the consumer to filter on a color and be presented a list of products in other colors. From a technical point of view it's clear it's "products having that color", but the average user is just confused.

**Comment by khouloudbelguith** [comment]
@Maboz, it could be an improvement issue, in the FO, when we search with attributes, specific image for each combination should be displayed & not the image of the combination by default.
@marionf @colinegin what do you think?
Thanks!

**Comment by colinegin** [comment]
@PierreRambaud do you think this could be an improvement in the faceted search module ?

**Comment by PierreRambaud** [comment]
Everything can be an improvement, we need to fix all bugs and refactor the code before adding anything :/

**Comment by marionf** [comment]
Related to #12241

**Comment by marionf** [comment]
If one filter is selected in FO (let's say Red), we should display the first image of this combination.
If many filters are selected in FO (let's say Blue and black) and a product has these 2 colors, we should display the first image of the default combination in product list.
If no filter are selected in FO, we should display the cover image of the product (issue #12241)

**Comment by kpodemski** [comment]
great idea

**Comment by colinegin** [comment]
@marionf as just discussed, the issue not only applies to the combination images, but also to the price which is not updated according to the filter.

**Comment by marionf** [comment]
Reported here: https://github.com/PrestaShop/PrestaShop/issues/15164

**Comment by itsmillertime65** [comment]
@marionf I see that this was removed from In progress, so does this mean it will be fixed in 1.7.7? Any way I can get this code and manually fix it for our PrestaShop site earlier? We have received many emails from customers saying they have not been able to find certain flavors of our products via search. These flavors are set up as combinations of a product and with the cover image not changing after searching or filtering for the flavor they are just assuming that we don't have that flavor.

**Comment by marionf** [comment]
@itsmillertime65 no, it will not be fixed for 1.7.7
We need to specify a lot of things before to do it

**Comment by itsmillertime65** [comment]
@marionf bummer.

**Comment by LollipopAdmin** [comment]
@marionf 
Any news on this suggestion from Maboz. I was looking at that feature too actually. 
I know it is available in some external module but to me this should be part of native faceted search module which is just great and would be even greater !!! 
I completely agree with the sentence of Maboz above: "t's quite confusing for the consumer to filter on a color and be presented a list of products in other colors. From a technical point of view it's clear it's "products having that color", but the average user is just confused." 
He clearly gets a point here. It woudl be so much better if filtering on a red color, you get all red products if any. 
I also understand the next questions: But what if you then click yellow... What do you display if both red an yellow exist.. 
I would go for the latest choice of the user or potentially an append..  like showing all red products and all yellow products... 
My 2 cents... 
Can you just let us know if this has any chance to be part of your backlog on day? 
Tks in advance

**Comment by jromero-onclick** [comment]
Are there any updates regarding this bug/improvement?

**Comment by Hlavtox** [comment]
If you use Prestashop 9, latest versions of ps_facetedsearch and enable a new option, it will show the combination image in the listing. :-)
